packages/api/src/hooks/p2p/useAdvertDelete.ts

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useMemo } from 'react';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TPayload = Parameters<ReturnType<typeof useMutation<'p2p_advert_update'>>['mutate']>[0]['payload'];
+
+/** A custom hook that deletes a P2P advert. This can only be used by an approved P2P advertiser.
+ * 
+ * To delete an advert, specify the advert ID to delete, for instance:
+ * @example
+ *  mutate({
+        "id": 1234
+    });
+ *
+ * Once this is mutated, the advert with ID of 1234 will be deleted.
+*/
+const useAdvertDelete = () => {
+    const invalidate = useInvalidateQuery();
+    const {
+        data,
+        mutate: _mutate,
+        ...rest
+    } = useMutation('p2p_advert_update', {
+        onSuccess: () => {
+            invalidate('p2p_advert_list');
+        },
+    });
+
+    const mutate = useCallback(
+        (payload: Omit<TPayload, 'delete'>) =>
+            _mutate({
+                payload: {
+                    ...payload,
+                    delete: 1,
+                },
+            }),
+        [_mutate]
+    );
+
+    const modified_data = useMemo(() => {
+        if (!data?.p2p_advert_update) return undefined;
+
+        return {
+            ...data?.p2p_advert_update,
```

Comment:
Can we add `data?.p2p_advert_update` into a variable? instead of repeating this multiple times?

Also we are checking if the data is undefined returns undefined so base on this we don't need to optional chain again 🤔